File: code/python/1432.py
# https://leetcode.com/problems/max-difference-you-can-get-from-changing-an-integer/
class Solution:
    def maxDiff(self, num: int) -> int:
        s = str(num)
        for ch in s:
            if ch != '9':
                num_max = int(s.replace(ch, '9'))
                break
        else:
            num_max = num
        
        if s[0] != '1':
            num_min = int(s.replace(s[0], '1'))
        else:
            for ch in s[1:]:
                if ch not in ('01'):
                    num_min = int(s.replace(ch, '0'))
                    break
            else:
                num_min = num
        return num_max - num_min

File: code/java/1432.java
// https://leetcode.com/problems/max-difference-you-can-get-from-changing-an-integer/
class Solution {
    public int maxDiff(int num) {
        String s = Integer.toString(num);
        int max = num, min = num;
        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (ch != '9') {
                max = Integer.parseInt(s.replace(ch, '9'));
                break;
            }
        }
        if (s.charAt(0) != '1')
            min = Integer.parseInt(s.replace(s.charAt(0), '1'));
        else
            for (int i = 1; i < s.length(); i++) {
                char ch = s.charAt(i);
                if (ch != '0' && ch != '1') {
                    min = Integer.parseInt(s.replace(ch, '0'));
                    break;
                }
            }
        return max - min;
    }
}
